hedera-mirror-rest/middleware/requestHandler.js

```diff
@@ -0,0 +1,63 @@
+/*-
+ * ‌
+ * Hedera Mirror Node
+ * ​
+ * Copyright (C) 2019 - 2020 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+'use strict';
+
+const qs = require('qs');
+
+const requestLogger = function (req, res, next) {
+  logger.debug(`Client: [ ${req.ip} ] URL: ${req.originalUrl}`);
+  return next();
+};
+
+/**
+ * Manage request query params to support case insensitive keys
+ * Express default query parer uses qs, other option is querystring, both are case sensitive
+ * Parse using default qs logic and use to populate a new map in which all keys are lowercased
+ * @param queryString
+ * @returns Query string map object
+ */
+const requestQueryParser = (queryString) => {
+  // parse first to benefit from qs query handling
+  const parsedQueryString = qs.parse(queryString);
+
+  const caseInsensitiveQueryString = {};
+  for (const key of Object.keys(parsedQueryString)) {
+    const lowerKey = key.toLowerCase();
+    const currentValue = caseInsensitiveQueryString[lowerKey];
+    if (currentValue) {
+      // handle repeated values. Add to array if applicable of convert to array
```

Comment:
nit: Think this is a typo.
```suggestion
      // handle repeated values. Add to array if applicable or convert to array
```
```suggestion
      // handle repeated values. Add to array if applicable of convert to array
```

Reply:
Fixed